### ISSUE TITLE
question regards to loss calculation

### ISSUE THREAD

**Issue by kwonmha** [icoxfog417/tying-wv-and-wc#2]
For augmented model, you added alpha(0.5) * temperature(10) * augmented_loss to ordinary loss.
How did you choose alpha and temperature?
And why did you inserted multiplying temperature to augmented loss?
Because it's not shown in the paper.
And have you tested using only augmented loss without adding it to ordinary loss?
I think it's not explicitly mentioned in paper.
TY

**Comment by icoxfog417** [comment]
Augmented loss is calculated as follows (p2, formula 3.3). 

![image](https://user-images.githubusercontent.com/544269/28866071-cf2cf272-77ac-11e7-9982-3dab820a77b3.png)

So we have to decide the parameter α. The author describes the way to compute α in the Appendix(p12). 

![image](https://user-images.githubusercontent.com/544269/28866032-b3bf2be0-77ac-11e7-8d8f-9d65c0a11900.png)

**α=γτ**. I implement this at [here](https://github.com/icoxfog417/tying-wv-and-wc/blob/master/model/augmented_model.py#L52).

**Comment by kwonmha** [comment]
Thank you for answer.
Maybe I've overlooked appendix part.

**Comment by kwonmha** [comment]
Thank you for answer.
Maybe I've overlooked the appendix part.

**Comment by kwonmha** [comment]
Sorry but I have another question to make certain.
How should I get output of the network for validating or testing?
Is it softmax( (Wh+b)/t )? 

And I think you didn't divide Wh+b by temperature when calculating cross entropy like eq(3.1) on paper.

**Comment by kwonmha** [comment]
You can ignore last 2 sentences of my previous comment.
I got confused.
Regards to the output of the network for validating or testing, it can be softmax(Wh+b), right?

**Comment by icoxfog417** [comment]
I think *softmax( (Wh+b)/t )* is used to calculate augmented loss only.
So network output will be *softmax(Wh+b)*.

**Comment by kwonmha** [comment]
I think so too.
Thank you!

**Comment by kwonmha** [comment]
It's a kind of glitch, but in formulation.png, it looks like you used softmax(Wh/t) to calculate both CE and KL.
Which may be not the case.
